### DISCUSSION TITLE
Why is `vue/multi-word-component-names` rule disabled also for components?

### DISCUSSION THREAD

**Discussion by stefanobartoletti** [nuxt/eslint-config#451] [Q&A]
I noticed that this configuration sets the `vue/multi-word-component-names` to `off`, and it affects the following patterns (basically everything):

```
app.{js,ts,jsx,tsx,vue}
error.{js,ts,jsx,tsx,vue}
layouts/**/*.{js,ts,jsx,tsx,vue}
pages/**/*.{js,ts,jsx,tsx,vue}
components/*/**/*.{js,ts,jsx,tsx,vue}
```

I know that `app.vue`, `error.vue`, layouts and pages are supposed to be single-word for good reasons, so it makes sense to disable this rule in these cases, but why is this turned off also for generic components in the `components` directory?

According to the Vue style guide, components are supposed to be multi-word, to prevent possible conflicts, is there a reason why Nuxt is setting a different standard?

Please note that I'm not asking to change this rule, or to give a different preset, I am already overriding this in my projects' configs.

I'm only asking for the sake of knowledge, what is the rationale behind this choice, because maybe I'm getting this wrong and I could remove the more strict rule that I'm using myself if there is a good reason for this.

Thanks!

**Comment by mkubenka** [comment]
I guess the main reason behind this decision is that by default Nuxt auto-import components based on path:

> https://nuxt.com/docs/guide/directory-structure/components#component-names

So, in default configuration is completely valid to have single-word components as Nuxt would translate it to multi-word during import.

**Comment by stefanobartoletti** [comment]
@mkubenka Yes, it looks like the rule is allowing single-name component only in subdirs of `components`, that effectively prefixes them with the dir name, so they are multi-word in the end.

Only components directly inside `components` are marked as errors, and it makes sense
